pkg/cri/opts/spec_linux.go

```diff
@@ -471,6 +472,10 @@ func WithResources(resources *runtime.LinuxContainerResources, tolerateMissingHu
 		if limit != 0 {
 			s.Linux.Resources.Memory.Limit = &limit
 		}
+		if swapLimit != 0 {
+			s.Linux.Resources.Memory.Swap = &swapLimit
```

Comment:
I'm not sure if we want to add any cgroupsv1 vs cgroupsv2 error handling here? - Or if that would be better elsewhere?

Reply:
This is the right implementation for v1 but for v2 if the memorylimit and swap limit are set, swap should be set to their difference (= 0 in the case that they are the same).

Also, you may want to validate this; swapLimit must be >= memory limit.

Reply:
It seems like this is already handled further down the chain (see tests, where we minus memory from the swap set here), gonna need to figure out cgroupsv1 testing separately

Reply:
yep, this is a good strategy, we will let the kernel validate and return the error if something is not correct.